modules/imaging_browser/php/NDB_Menu_Filter_imaging_browser.class.inc

```diff
@@ -95,30 +115,36 @@ class NDB_Menu_Filter_Imaging_Browser extends NDB_Menu_Filter
             JOIN files f ON (f.SessionID=s.ID) 
             LEFT JOIN files_qcstatus fqc ON (fqc.FileID=f.FileID) 
             JOIN mri_acquisition_dates md ON (md.SessionID=s.ID)
-            LEFT JOIN (SELECT files.SessionID, MIN(QCLastChangeTime)
-            as QCLastChangeTime FROM files
-            LEFT JOIN files_qcstatus USING (FileID)
+            LEFT JOIN (SELECT files.SessionID, MIN(QCLastChangeTime) 
+            as QCLastChangeTime FROM files 
+            LEFT JOIN files_qcstatus USING (FileID) 
             WHERE OutputType='native' AND AcquisitionProtocolID 
             NOT IN (1, 2, 3, 52) GROUP BY files.SessionID) nd 
-            ON (nd.SessionID=f.SessionID)
-            LEFT JOIN (SELECT files.SessionID,
-            MIN(files_qcstatus.QCStatus+0) as T1QC FROM files 
-            JOIN files_qcstatus USING (FileID)
-            WHERE files.AcquisitionProtocolID=44 
-            AND files_qcstatus.QCStatus IN (1, 2)
-            GROUP BY files.SessionID) t1pass ON (t1pass.SessionID=f.SessionID)
+            ON (nd.SessionID=f.SessionID) ";
+
+        $left_join = "";
+        foreach ($case_desc as $key => $value) {
+            $left_join .= "
             LEFT JOIN (SELECT files.SessionID, 
-            MIN(files_qcstatus.QCStatus+0) as T2QC FROM files 
-            JOIN files_qcstatus USING (FileID)
-            WHERE files.AcquisitionProtocolID=45 
-            AND files_qcstatus.QCStatus IN (1, 2)
-            GROUP BY files.SessionID) t2pass ON (t2pass.SessionID=f.SessionID)
+            MIN(files_qcstatus.QCStatus+0) as " . $DB->quote($qc[$key]) . " 
+            FROM files 
+            JOIN files_qcstatus USING (FileID) 
+            WHERE files.AcquisitionProtocolID=" . $DB->quote($key) . " 
+            AND files_qcstatus.QCStatus IN (1, 2) 
+            GROUP BY files.SessionID) `" . $pass[$key] . "` 
+            ON (`" . $pass[$key] . "`.SessionID=f.SessionID) ";
+        }
+
+        $where = "
             WHERE 
             s.Active = 'Y' AND
             f.PendingStaging=0 AND 
             f.FileType='mnc' AND 
             f.AcquisitionProtocolID not in (1, 2, 3, 52)";
 
+        $this->query .= $left_join;
```

Comment:
It's nit picky, but this would read better if the variable was named "left_joins"